From 339c8704c1aa0abf206f7ab99674d4e671b43472 Mon Sep 17 00:00:00 2001
From: Eran Ben-Avi <benavi@marvell.com>
Date: Thu, 11 Jul 2013 18:13:13 +0300
Subject: [PATCH 0813/1825] Add FIQ exception counter: cat /proc/fiq_dump

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit cdc3392789a4c67b9d16e99298c9b96a6f78cdd5

	Code imported from cid:
	46eb9e0ba83ee674fda146658d9c7b395bbcd103

Change-Id: Iaafb836893caa055724400c0fbb2dad03330d9e6
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Signed-off-by: Eran Ben-Avi <benavi@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/plat-armada/mv_drivers_lsp/mv_proc/proc.c |   27 +++++++++++++++++++-
 1 files changed, 26 insertions(+), 1 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_proc/proc.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_proc/proc.c
index 7502392..609cde3 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_proc/proc.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_proc/proc.c
@@ -21,7 +21,6 @@
 #include <linux/device.h>
 #include <linux/slab.h>
 #include <linux/string.h>
-#include <linux/sysdev.h>
 #include <linux/proc_fs.h>
 #include <linux/version.h>
 #include <asm/io.h>
@@ -370,6 +369,32 @@ int __init start_resource_dump(void)
 
 module_init(start_resource_dump);
 
+#ifdef CONFIG_MV_PHONE_USE_FIQ_PROCESSING
+/*
+ * FIQ proc entry
+ */
+static struct proc_dir_entry *fiq_proc;
+extern int fiq_cnt;
+
+int fiq_proc_read(char *buffer, char **buffer_location, off_t offset,
+		  int buffer_length, int *zero, void *ptr)
+{
+	if (offset > 0)
+		return 0;
+	return sprintf(buffer, "%d\n", fiq_cnt);
+}
+
+int __init fiq_proc_init(void)
+{
+	fiq_proc = create_proc_entry("fiq_dump", 0444, NULL);
+	fiq_proc->read_proc = fiq_proc_read;
+	fiq_proc->write_proc = NULL ; /* fiq_proc_write; */
+	fiq_proc->nlink = 1;
+	return 0;
+}
+module_init(fiq_proc_init);
+#endif /* CONFIG_MV_PHONE_USE_FIQ_PROCESSING */
+
 /* global variables from 'regdump' */
 static struct proc_dir_entry *soc_type;
 static struct proc_dir_entry *board_type;
-- 
1.7.5.4

